tests/comparisons/Model/testBakeTableValidation.php

```diff
@@ -44,32 +44,33 @@ public function validationDefault(Validator $validator)
     {
         $validator
             ->numeric('id')
-            ->allowEmpty('id', 'create');
+            ->allowEmptyString('id', 'create');
 
         $validator
             ->scalar('name')
+            ->maxLength('name', 100, 'Name must be shorter than 100 characters.')
             ->requirePresence('name', 'create')
-            ->notEmpty('name')
-            ->maxLength('name', 100, 'Name must be shorter than 100 characters.');
+            ->notEmptyString('name');
 
         $validator
             ->nonNegativeInteger('count')
             ->requirePresence('count', 'create')
-            ->notEmpty('count');
+            ->notEmptyString('count');
```

Comment:
string for ints is a bit weird.
can we use ...Scalar here etc?

Reply:
From what I remember the thinking was that POST data from forms is always strings.

Reply:
But we know that this is not true for this validation process. Marshalled data in general can also come from webservice, and other endpoints, as such they sure can be in a nonstring format already.

Reply:
Yes data from webservices can be in non string format but if a value of valid type is set for a non string field then it is non empty anyway and "not empty" check will pass as expected.

Taking the example of an integer field only way to indicate empty value is passing `null` or empty string for it, which is covered by using `notEmptyString()`. (All `notEmpty*` methods treat `null` as empty as expected).

Reply:
I didnt mean null, but but the actual value (int, float, ..).
So for me scalar still sounds more correct than string.
We might have to patch the core here.

Reply:
Bake can be updated if/when core is updated. For now this is good to go.